README.md

```diff
@@ -1,66 +1,12 @@
-## Foundry
-
-**Foundry is a blazing fast, portable and modular toolkit for Ethereum application development written in Rust.**
-
-Foundry consists of:
-
--   **Forge**: Ethereum testing framework (like Truffle, Hardhat and DappTools).
--   **Cast**: Swiss army knife for interacting with EVM smart contracts, sending transactions and getting chain data.
--   **Anvil**: Local Ethereum node, akin to Ganache, Hardhat Network.
--   **Chisel**: Fast, utilitarian, and verbose solidity REPL.
-
-## Documentation
-
-https://book.getfoundry.sh/
-
-## Usage
-
-### Build
-
-```shell
-$ forge build
-```
-
-### Test
-
-```shell
-$ forge test
-```
-
-### Format
-
-```shell
-$ forge fmt
-```
-
-### Gas Snapshots
-
-```shell
-$ forge snapshot
-```
-
-### Anvil
-
-```shell
-$ anvil
-```
-
-### Deploy
-
-```shell
-$ forge script script/Counter.s.sol:CounterScript --rpc-url <your_rpc_url> --private-key <your_private_key>
-```
-
-### Cast
-
-```shell
-$ cast <subcommand>
-```
-
-### Help
-
-```shell
-$ forge --help
-$ anvil --help
-$ cast --help
-```
+# 🏛️⚡️ Spark Gov Relay 🏛️⚡️
+This repository contains infrastructure necessary to process cross-chain execution of governance proposals.
+
+The codebase uses [Foundry](https://github.com/foundry-rs/foundry) as the development framework. In order to run tests, deploy contracts, or perform other operations, use standard [Foundry](https://github.com/foundry-rs/foundry) commands.
+
+## ⚙️ Components
+### 🔊 Forwarders
+Forwarders are contracts on the host domain, where the governance of the protocol resides. <br>These contracts abstract away all complexity of using a bridge and facilitate seamless message passing between domains. <br>Host domain admin uses forwarders to trigger payload executions on a bridged domain instances of the protocol.
+### 🚦 Executors
+Executors serve as admins of the bridged domain instances of the protocol. <br>They are responsible for storying the queue of proposals passed from the host domain governance and their execution. <br> They manage bridged domain protocol instance using standard payload pattern.
+## ✍️ Architecture Diagram
+![Architecture Diagram](/diagram.png)
```

Comment:
This diagram is not general. The validator set is specific to Gnosis Chain and this does not necessarily exist for other chains such as L2s.

Reply:
There has to be some EOA-ish type of entity that initiates the execution on the L2 side and that's what I wanted to picture here. I'm in favour of renaming and keeping it, but I can remove as well. Any suggestions?

Reply:
I would call it "Relayer" then or remove it.

Reply:
Ok, I removed that part actually.